### PR TITLE
Use mkstemp instead of tmpname when available

### PR DESCRIPTION
**Pull request by samuel40791765**
### Description of changes: 
Despite silencing tmpname deprecation warnings, the warning still appears on Linux platforms when compiling tests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: ssl/ssl_test.cc [hunk 1]
```diff
@@ -11852,10 +11848,22 @@ TEST(SSLTest, SSLFileTests) {
...
+  OPENSSL_END_ALLOW_DEPRECATED
+#else
+  char rsa_pem_filename[] = "/tmp/fileXXXXXX";
+  char ecdsa_pem_filename[] = "/tmp/fileXXXXXX";
+  ASSERT_TRUE(mkstemp(rsa_pem_filename));
+  ASSERT_TRUE(mkstemp(ecdsa_pem_filename));
+#endif
 
   ScopedFILE rsa_pem(fopen(rsa_pem_filename, "w"));
   ScopedFILE ecdsa_pem(fopen(ecdsa_pem_filename, "w"));
```
Comment (WillChilds-Klein) [inline ssl/ssl_test.cc:11869]: If we don't care about the file name, can't you just use  `tmpfile()` [as in bio_test.cc](https://github.com/aws/aws-lc/blob/main/crypto/bio/bio_test.cc#L176)? It should be [portable enough](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/tmpfile?view=msvc-170). These lines would just become:

```
ScopedFile rsa_pem(tmpfile());
ScopedFile ecdsa_pem(tmpfile());
```

**Comment by justsmth** [comment]
Another option is `tempnam` (a [POSIX.1-2008 alternative](https://pubs.opengroup.org/onlinepubs/9699919799/functions/tempnam.html) that's thread-safe?) that appears to be supported on [Windows](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/tempnam?view=msvc-160), [Linux](https://www.man7.org/linux/man-pages/man3/tempnam.3.html) and [Mac](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/tempnam.3.html)

**Review by justsmth — APPROVED** [review]

File: ssl/ssl_test.cc (hunk 1 above)
Comment (samuel40791765) [reply ssl/ssl_test.cc:11869]: That's the conundrum I've been trying to solve with these tests :( 
`SSL_CTX_use_certificate_chain_file` and `SSL_use_certificate_chain_file` directly take the file name as the second argument and parse the certificates for the user. `tmpfile` was the initial plan, but it doesn't provide an interface to extract the created file name.

**Review by samuel40791765 — COMMENTED** [review]
> Another option is tempnam (a [POSIX.1-2008 alternative](https://pubs.opengroup.org/onlinepubs/9699919799/functions/tempnam.html) that's thread-safe?) that appears to be supported on [Windows](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/tempnam?view=msvc-160), [Linux](https://www.man7.org/linux/man-pages/man3/tempnam.3.html) and [Mac](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/tempnam.3.html)

Sure, `tempname` has the same security and deprecation warning as `tmpnam`, but we can opt for the additional tread-safety on Windows.

**Review by justsmth — APPROVED** [review]

File: ssl/ssl_test.cc (hunk 1 above)
Comment (WillChilds-Klein) [reply ssl/ssl_test.cc]: Ah, I see. That makes sense.

**Review by WillChilds-Klein — APPROVED** [review]

**Pull request merged**